sleepio/community-app-web
# PG-1377, PG-1535, PG-1563: Logout and redirection + Bugfixes

Redirect the user to sleepios login page when not logged in to sleepio


**Reminder to create the PR against this repository, and not the base fork of `rafalp/Misago`.**

### JIRA Ticket

https://bighealth.atlassian.net/browse/PG-1377
https://bighealth.atlassian.net/browse/PG-1535

### Problem

If a user's sleepio session ends, or doesn't exist, they should be redirected from Community back to Sleepio for login

### Solution

We have enriched the authentication middleware to handle these redirects. We use django's `reverse` function to pull urls from the view names, rather than be explicit.

We have also excluded the `/admincp` path from custom token validation, so we can continue to log in as admins.

### Test Plan

#### Unit tests + Integration Tests

* Updated test_middleware.py to validate redirects and path exclusions

#### Manual Tests

*PG-1377 test*
* navigating directly to community without a valid Sleepio session results in redirection to the Sleepio Login Page
* after a successful sign out from Sleepio, interacting with any open community tab results in redirection to the Sleepio login page
* after a successful sign out from Sleepio, refreshing any open community tab results in redirection to the Sleepio login page
* after a successful sign out from Sleepio, the automatic session polling results in redirection to the Sleepio login page
* admins can continue to access the admin portal at {communityUrl}/admincp, log in via that portal, and engage with both the admin dashboard and the community message board

*PG-1535 test*
_should be run with users that have never used community before_
* U1 log into sleepio
* U1 open community, close the community tab
* U1 log out of sleepio
* U2 log into sleepio
* U2 open community, confirm account is created for U2.

### Rollout


#### Related PRs

Depends on https://github.com/sleepio/community-app-web/pull/9
